osfoffline/utils/authentication.py

```diff
@@ -0,0 +1,196 @@
+import asyncio
+import datetime
+import furl
+import logging
+
+import aiohttp
+import bcrypt
+from PyQt5.QtWidgets import QMessageBox
+
+from osfoffline import settings
+from osfoffline.database_manager.db import session
+from osfoffline.database_manager.models import User
+from osfoffline.database_manager.utils import save
+from osfoffline.polling_osf_manager.remote_objects import RemoteUser
+
+
+def generate_hash(password):
+    """ Generates a password hash using `bcrypt`.
+        Number of rounds for salt generation is 12.
+
+    :return: hashed password
+    """
+    return bcrypt.hashpw(
+        password.encode('utf-8'),
+        bcrypt.gensalt(12)
+    )
+
+token_request_body = json.loads(
+    '{"data": {"type": "tokens", "attributes": {"name": "OSF-Offline {}", "scopes": "osf.full_write"}}}'.format(datetime.date.today())
```

Comment:
Since you're writing the dictionary inline, no need to convert it to a string. Just specify it as a dictionary to start.